.github/workflows/workflow.yml

```
@@ -0,0 +1,46 @@
+name: D2L License Checker CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: Test
+    timeout-minutes: 2
+    strategy:
+      matrix:
+        node: ['10', '12', '14', '15']
+    steps:
+      - name: Checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+      - name: Setup Node
+        uses: Brightspace/third-party-actions@actions/setup-node
+        with:
+          node-version: ${{ matrix.node }}
+      - name: npm install
+        run: npm ci
+      - name: lint
+        run: npm run lint
+      - name: test
+        run: npm run test
+
+  release:
+    if: github.ref == 'refs/heads/master'
```

```suggestion
    if: github.ref == 'refs/heads/master'
    needs: test
```

I think. Otherwise, `release` will run before `test`.

Alternatively, you can just move the `Semantic Release` step to the last step of `test`.

---

I can't move the Semantic release step up because it's now running on 4 node versions, meaning (if I understand correctly) that it would run the semantic release 4 times.

---

Oh right. You could do something like this to pick the version you want:

```yaml
if: ${{ matrix.node == 14 }}
```

Not a big deal either way I think.

---

I think I like the `needs` method a bit more, it's some yaml duplication but I like how explicit it looks to someone who has never read the workflow before.